### PR TITLE
Expose Highest Watermark and Newest Offset via an RPC endpoint

### PR DESCRIPTION
**Pull request by LaPetiteSouris**
Refer to [issue](https://github.com/liftbridge-io/liftbridge/issues/111).

This PR is the prerequisite for [this](https://github.com/liftbridge-io/liftbridge/pull/242)

Change logs:

- Add `FetchPartitionMetadata` endpoint
- Add `FetchPartitionMetadata` request and response message accordingly

### CODE REVIEW

File: api.proto [hunk 1]
```diff
@@ -107,6 +107,19 @@ message FetchMetadataResponse {
     repeated StreamMetadata metadata = 2; // Information for all streams
 }
 
+// FetchPartitionMetadataRequest is sent to retrieve latest partition metadata.
+// Mainly it is useful to retrieve Highest Watermark and Highest Offset of the partition leader
+message FetchPartitionMetadataRequest {
+    string stream       = 1;
+    int32  partitionID  = 2;
```
Comment (tylertreat) [inline api.proto]:
```suggestion
    int32  partition  = 2;
```

For consistency with other parts of the API.

File: api.proto [hunk 2]
```diff
@@ -107,6 +107,19 @@ message FetchMetadataResponse {
...
+}
+
+
+//  FetchPartitionMetadataResponse contains the partition metadata requested
+message FetchPartitionMetadataResponse {
+    PartitionMetadata partitionMetadata =1;
```
Comment (tylertreat) [inline api.proto]:
```suggestion
    PartitionMetadata metadata = 1;
```

**Review by tylertreat — CHANGES_REQUESTED** [review]
Just a couple minor tweaks but otherwise I think this can go in.

File: api.proto (hunk 1 above)
Comment (LaPetiteSouris) [reply api.proto]: done

File: api.proto (hunk 2 above)
Comment (LaPetiteSouris) [reply api.proto]: done

**Pull request merged**